modules/fedimint-lnv2-client/src/send_sm.rs

```diff
@@ -0,0 +1,338 @@
+        global_context: DynGlobalClientContext,
+        txid: TransactionId,
+    ) -> Result<(), String> {
+        global_context.await_tx_accepted(txid).await
+    }
+
+    async fn transition_funding(
+        result: Result<(), String>,
+        old_state: SendStateMachine,
+    ) -> SendStateMachine {
+        let funding = match old_state.state {
+            SendSMState::Funding(funding) => funding,
+            _ => panic!("Unexpected prior state"),
+        };
+
+        SendStateMachine {
+            operation_id: old_state.operation_id,
+            contract: old_state.contract,
+            state: match result {
+                Ok(()) => SendSMState::Funded(Funded {
+                    gateway_api: funding.gateway_api,
+                    invoice: funding.invoice,
+                    refund_keypair: funding.refund_keypair,
+                }),
+                Err(error) => SendSMState::Rejected(error),
+            },
+        }
+    }
+
+    async fn gateway_send_payment(
+        gateway_api: SafeUrl,
+        federation_id: FederationId,
+        contract: OutgoingContract,
+        invoice: Bolt11Invoice,
+    ) -> Result<[u8; 32], Signature> {
+        loop {
+            match Self::try_gateway_send_payment(
+                gateway_api.clone(),
+                federation_id,
+                contract.clone(),
+                invoice.clone(),
+            )
+            .await
+            {
+                Ok(gateway_response) => match gateway_response {
+                    Ok(gateway_response) => {
+                        if contract.verify_gateway_response(&gateway_response) {
+                            return gateway_response;
+                        } else {
+                            error!("Invalid gateway response: {gateway_response:?}");
+                        }
+                    }
+                    Err(error) => {
+                        error!("Gateway returned error: {error}")
+                    }
+                },
+                Err(error) => {
+                    error!("Error while trying to reach gateway: {error}");
+                }
+            }
+
+            sleep(RETRY_DELAY).await;
+        }
+    }
+
+    async fn try_gateway_send_payment(
+        gateway_api: SafeUrl,
+        federation_id: FederationId,
+        contract: OutgoingContract,
+        invoice: Bolt11Invoice,
+    ) -> anyhow::Result<Result<Result<[u8; 32], Signature>, String>> {
+        let result = reqwest::Client::new()
+            .post(
+                gateway_api
+                    .join("send_payment")
+                    .expect("'send_payment' contains no invalid characters for a URL")
+                    .as_str(),
+            )
+            .json(&SendPaymentPayload {
+                federation_id,
+                contract,
+                invoice,
+            })
+            .send()
+            .await?
+            .json::<Result<Result<[u8; 32], Signature>, String>>()
```

Comment:
Maybe we can newtype `Result<[u8; 32], Signature>`. Seeing this return a Result of a Result is a little weird

Reply:
(but it makes sense knowing the the forfeit signature is returned on failure)